sdk/src/main/java/com/mapbox/maps/ViewAnnotationManagerImpl.kt

```diff
@@ -38,6 +39,7 @@ internal class ViewAnnotationManagerImpl(
   internal val idLookupMap = ConcurrentHashMap<View, String>()
   private val currentlyDrawnViewIdSet = mutableSetOf<String>()
   private val unpositionedViews = mutableSetOf<View>()
+  private val annotationOptionsLookupMap = ConcurrentHashMap<View, ViewAnnotationOptions>()
```

Comment:
Is it okay to keep a local copy of `ViewAnnotationOptions` instead of calling the [native method](https://github.com/mapbox/mapbox-maps-android/blob/main/sdk/src/main/java/com/mapbox/maps/MapboxMap.kt#L1774) to get those?

Could there be some discrepancies between the one stored here and the one in the native side?

We could build that map when `annotations` are requested by using that native method plus the `idLookupMap` that we already have.

Reply:
> Is it okay to keep a local copy of ViewAnnotationOptions instead of calling the [native method](https://github.com/mapbox/mapbox-maps-android/blob/main/sdk/src/main/java/com/mapbox/maps/MapboxMap.kt#L1774) to get those?

we do update these annotationOptions in map whenever `updateViewAnnotation` is called. so there will not be any discrepancies between the ones stored vs native one.

> We could build that map when annotations are requested by using that native method plus the idLookupMap that we already have.

I guess yes, but we don't want to call native method that many times (that's the idea initially ), for eg if there are 100 viewAnnotations added and user request for annotations map, then we would call [native method](https://github.com/mapbox/mapbox-maps-android/blob/main/sdk/src/main/java/com/mapbox/maps/MapboxMap.kt#L1774) that many times every time user query annotations. I think [this](https://github.com/mapbox/mapbox-maps-android/pull/1751#discussion_r1006655113) is similar question.

Reply:
Indeed is the same question, sorry about that.
Thanks for explaining!

Reply:
@jush updated implementation to build map from idlookup.